### PR TITLE
fix: images in shared news are not displayed - EXO-61176

### PR DESCRIPTION
**Pull request by GouadriaHanen**
before this change, images are not accessible from shared news of space I'm not a member of, since the user doesn't have the right to see them
after this change, the images are displayed successfully after adding the read permission for the shared news images

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java [hunk 1]
```diff
@@ -1267,5 +1271,18 @@ private Session getSession(SessionProvider sessionProvider) throws RepositoryExc
                                                        .getDefaultWorkspaceName(),
                                       repositoryService.getCurrentRepository());
   }
-  
+
+  private  void updateNewsImagesPermissions(News news ,  SessionProvider sessionProvider, Space space  ) throws RepositoryException {
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java]:
```suggestion
  private  void updateNewsImagesPermissions(News news, SessionProvider sessionProvider, Space space) throws RepositoryException {
```

File: services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java [hunk 2]
```diff
@@ -1267,5 +1271,18 @@ private Session getSession(SessionProvider sessionProvider) throws RepositoryExc
...
+    Matcher matcher = IMAGE_SRC_PATTERN.matcher(news.getBody());
+    while (matcher.find()) {
+      String match = matcher.group(1);
+      String imageUUID = match.substring(match.lastIndexOf("/") + 1);
+      ExtendedNode image = (ExtendedNode) getNodeById(imageUUID, sessionProvider);
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java:1280]: Could you do a check here if the content is really present or not ?

File: services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java [hunk 3]
```diff
@@ -1267,5 +1271,18 @@ private Session getSession(SessionProvider sessionProvider) throws RepositoryExc
...
+      if (image.canAddMixin(EXO_PRIVILEGEABLE)) {
+        image.addMixin(EXO_PRIVILEGEABLE);
+      }
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java]: Is there a need to add this mixin ? you already loaded the image by UUID ?

File: services/src/test/java/org/exoplatform/news/storage/jcr/JcrNewsStorageTest.java [hunk 4]
```diff
@@ -8,11 +8,8 @@
 import java.util.*;
 
 import javax.jcr.*;
-import javax.jcr.query.QueryManager;
```
Comment (ahamdi) [inline services/src/test/java/org/exoplatform/news/storage/jcr/JcrNewsStorageTest.java:11]: Please add a new test for the new change ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java (hunk 3 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java]: we could not set the permission without adding EXO_PRIVILEGEABLE

File: services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java (hunk 2 above)
Comment (GouadriaHanen) [reply services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java:1280]: done

File: services/src/test/java/org/exoplatform/news/storage/jcr/JcrNewsStorageTest.java (hunk 4 above)
Comment (GouadriaHanen) [reply services/src/test/java/org/exoplatform/news/storage/jcr/JcrNewsStorageTest.java:11]: done

File: services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java (hunk 3 above)
Comment (ahamdi) [reply services/src/main/java/org/exoplatform/news/storage/jcr/JcrNewsStorage.java]: Yes you are right, I misread it

**Review by ahamdi — APPROVED** [review]

**Pull request merged**